src/code/fv-components/end-level-dialog.js

```diff
@@ -19,10 +19,9 @@ const EndLevelDialogView = ({challengeAwards, onNextChallenge, onTryAgain}) => {
       mission = challengeAwards.routeSpec.mission,
       challenge = challengeAwards.routeSpec.challenge,
       challengeCount = challengeAwards.challengeCount,
-      progress = challengeAwards.progress;
+      progress = challengeAwards.progress,
+      currentScore = getScoreValue(challengeAwards.currentScore);
```

Comment:
This is the fix for showing the correct gem for a completed challenge - even if it is worse than the "better" gem saved in your progress. The solution was to calculate the current score based off your current moves, rather than use the score in your progress, which doesn't change if your new attempt is worse.